assets/cms/js/vue/Manager.js

```diff
@@ -0,0 +1,80 @@
+/* eslint-disable no-new */
+/* global Vue */
+import Context from './Context'
+
+class Manager {
+    /**
+     * Ensures that our Concrete.Vue manager is available on the window object.
+     * Note: Do NOT call this before the global Concrete object is created in the CMS context.
+     *
+     * @param window
+     */
+    static bindToWindow(window) {
+        if (!window.Concrete) {
+            window.Concrete = {}
+        }
+        if (!window.Concrete.Vue) {
+            window.Concrete.Vue = new Manager()
+            window.Concrete.Vue.contexts = []
```

Comment:
Why not make this an object and have the context names be the key? Then you can access the defined contexts easier and be sure you don't have duplicates

Reply:
Also we should consider firing an event on the window like `concrete.vue.ready` so that people can hook into this without needing to wait for the entire dom to load